packages/core/src/linker/destroy_ref.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {EnvironmentInjector, inject} from '../di';
+import {LView} from '../render3/interfaces/view';
+import {getLView} from '../render3/state';
+import {storeLViewOnDestroy} from '../render3/util/view_utils';
+
+/**
+ * DestroyRef represents a concept of lifecycle scope where destroy callbacks can be registered.
```

Comment:
```suggestion
 * `DestroyRef` lets you set callbacks to run when a component or directive is destroyed.
```

I would make the description a bit more practically-oriented

Reply:
Let's chat about this one. We definitively should include injectors here as this is the main point of this API - to unify the concept of "cleanup" across components and injectors. Also, not sure about mentioning component / directive destruction: it is really about a view boundary.

Reply:
My motivation here is that the bit "concept of a lifecycle" is odd/confusing. 

Maybe something roughly like
```markdown
`DestroyRef` lets you set callbacks to run for any cleanup or destruction behavior.
The scope of this destruction depends on where `Destroy` is injected. If `DestroyRef`
is injected in a component or directive, the callbacks run when that component or
directive is destroyed. If <other case>
```

Reply:
👍